### PR TITLE
Bump baseline

### PR DESCRIPTION
**Pull request by timja**
Initial work on configuration as code support

98% of users who are upgrading their plugin version are on this version or above:
http://stats.jenkins.io/pluginversions/azure-vm-agents.html

(I haven't run the integration tests, but the unit tests pass)

### CODE REVIEW

**Comment by timja** [comment]
@gavinfish could you take a look please?

File: pom.xml [hunk 1]
```diff
@@ -123,41 +178,7 @@
...
-            <version>4.5</version>
-        </dependency>
-        <dependency>
-            <groupId>org.jenkins-ci.main</groupId>
-            <artifactId>jenkins-test-harness</artifactId>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.jenkins-ci.main</groupId>
-            <artifactId>jenkins-core</artifactId>
```
[the remark this answers is not in the document]
Comment (timja) [reply pom.xml]: not needed anymore, tests passed locally without them.

**Comment by timja** [comment]
build fixed @gavinfish

File: src/main/java/com/microsoft/azure/vmagent/remote/AzureVMAgentSSHLauncher.java [hunk 2]
```diff
@@ -104,6 +104,12 @@ public void launch(SlaveComputer agentComputer, TaskListener listener) {
         boolean successful = false;
         Session session = null;
 
+        SlaveComputer slaveComputer = agent.getComputer();
+        if (slaveComputer == null) {
+            LOGGER.log(Level.SEVERE, "AzureVMAgentSSHLauncher: launch: Got null computer.");
+            handleLaunchFailure(agent, Constants.AGENT_POST_PROV_AUTH_FAIL);
```
Comment (gavinfish) [inline src/main/java/com/microsoft/azure/vmagent/remote/AzureVMAgentSSHLauncher.java]: Should change failure message here, not a auth failure.

Comment (timja) [reply src/main/java/com/microsoft/azure/vmagent/remote/AzureVMAgentSSHLauncher.java]: I've pushed another message, feel free to amend it as you wish

**Pull request merged**